src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Split.cs

```diff
@@ -0,0 +1,264 @@
+            public bool MoveNext()
+            {
+                // Search for the next separator index.
+                int separatorIndex, separatorLength;
+                switch (_splitMode)
+                {
+                    case SpanSplitEnumeratorMode.None:
+                        return false;
+
+                    case SpanSplitEnumeratorMode.SingleElement:
+                        separatorIndex = _span.Slice(_startNext).IndexOf(_separator);
+                        separatorLength = 1;
+                        break;
+
+                    case SpanSplitEnumeratorMode.Any:
+                        separatorIndex = _span.Slice(_startNext).IndexOfAny(_separatorBuffer);
+                        separatorLength = 1;
+                        break;
+
+                    case SpanSplitEnumeratorMode.Sequence:
+                        separatorIndex = _span.Slice(_startNext).IndexOf(_separatorBuffer);
+                        separatorLength = _separatorBuffer.Length;
+                        break;
+
+                    case SpanSplitEnumeratorMode.EmptySequence:
+                        separatorIndex = -1;
+                        separatorLength = 1;
+                        break;
+
+                    default:
+                        Debug.Assert(_splitMode == SpanSplitEnumeratorMode.SearchValues, $"Unknown split mode: {_splitMode}");
+                        separatorIndex = _span.Slice(_startNext).IndexOfAny(_searchValues);
+                        separatorLength = 1;
+                        break;
+                }
+
+                _startCurrent = _startNext;
+                if (separatorIndex >= 0)
+                {
+                    _endCurrent = _startCurrent + separatorIndex;
+                    _startNext = _endCurrent + separatorLength;
+                }
+                else
+                {
+                    _startNext = _endCurrent = _span.Length;
+
+                    // Set _splitMode to None so that subsequent MoveNext calls will return false.
+                    _splitMode = SpanSplitEnumeratorMode.None;
+                }
+
+                return true;
+            }
+        }
+
+        /// <summary>Gets a <see cref="SearchValues{Char}"/> for all of the Unicode whitespace characters</summary>
+        private static SearchValues<char> WhiteSpaceSearchValues
+        {
+            get
+            {
+                return s_whiteSpaceSearchValues ?? Initialize();
+
+                [MethodImpl(MethodImplOptions.NoInlining)]
+                static SearchValues<char> Initialize()
+                {
+                    SearchValues<char> sv = SearchValues.Create("\t\n\v\f\r\u0020\u0085\u00a0\u1680\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u2028\u2029\u202f\u205f\u3000");
```

Comment:
Is there no constant for this anywhere?

Reply:
No, we don't use it anywhere else, to my knowledge.

Reply:
Did you whip this up from memory then? 😅 Should each of these be exhaustively tested? To lock in behaviour of what is considered `whitespace`?

Reply:
> Did you whip this up from memory then?

Nope:
```C#
for (int i = 0; i <= char.MaxValue; i++)
    if (char.IsWhiteSpace((char)i))
        Console.Write($"\\u{i:X4}");
```

> Should each of these be exhaustively tested?

The set is validated by the debug asserts immediately after this line. It should always be the same as char.IsWhiteSpace.

Reply:
I see, if that suffices from a testing perspective then fair! I guess since pipeline runs target release and debug this should be fine 👍